server/src/main/java/org/elasticsearch/common/geo/LuceneGeometriesUtils.java

```diff
@@ -0,0 +1,419 @@
+     **/
+    public static org.apache.lucene.geo.Line toLatLonLine(Line line) {
+        return toLatLonLine(line, IDENTITY, IDENTITY);
+    }
+
+    private static org.apache.lucene.geo.Line toLatLonLine(Line line, DoubleFunction latFunction, DoubleFunction lonFunction) {
+        return new org.apache.lucene.geo.Line(quantizeLats(line.getLats(), latFunction), quantizeLons(line.getLons(), lonFunction));
+    }
+
+    /**
+     * Transform an Elasticsearch {@link Polygon} into a lucene {@link org.apache.lucene.geo.Polygon}
+     **/
+    public static org.apache.lucene.geo.Polygon toLatLonPolygon(Polygon polygon) {
+        return toLatLonPolygon(polygon, IDENTITY, IDENTITY);
+    }
+
+    private static org.apache.lucene.geo.Polygon toLatLonPolygon(Polygon polygon, DoubleFunction latFunction, DoubleFunction lonFunction) {
+        org.apache.lucene.geo.Polygon[] holes = new org.apache.lucene.geo.Polygon[polygon.getNumberOfHoles()];
+        for (int i = 0; i < holes.length; i++) {
+            holes[i] = new org.apache.lucene.geo.Polygon(
+                quantizeLats(polygon.getHole(i).getY(), latFunction),
+                quantizeLons(polygon.getHole(i).getX(), lonFunction)
+            );
+        }
+        return new org.apache.lucene.geo.Polygon(
+            quantizeLats(polygon.getPolygon().getY(), latFunction),
+            quantizeLons(polygon.getPolygon().getX(), lonFunction),
+            holes
+        );
+
+    }
+
+    /**
+     * Transform an Elasticsearch {@link Rectangle} into a lucene {@link org.apache.lucene.geo.Rectangle}
+     **/
+    public static org.apache.lucene.geo.Rectangle toLatLonRectangle(Rectangle rectangle) {
+        return toLatLonRectangle(rectangle, IDENTITY, IDENTITY);
+    }
+
+    private static org.apache.lucene.geo.Rectangle toLatLonRectangle(Rectangle r, DoubleFunction latFunction, DoubleFunction lonFunction) {
+        return new org.apache.lucene.geo.Rectangle(
+            latFunction.apply(r.getMinLat()),
+            latFunction.apply(r.getMaxLat()),
+            lonFunction.apply(r.getMinLon()),
+            lonFunction.apply(r.getMaxLon())
+        );
+    }
+
+    /**
+     * Transform an Elasticsearch {@link Circle} into a lucene {@link org.apache.lucene.geo.Circle}
+     **/
+    public static org.apache.lucene.geo.Circle toLatLonCircle(Circle circle) {
+        return toLatLonCircle(circle, IDENTITY, IDENTITY);
+    }
+
+    private static org.apache.lucene.geo.Circle toLatLonCircle(Circle circle, DoubleFunction latFunction, DoubleFunction lonFunction) {
+        return new org.apache.lucene.geo.Circle(
+            latFunction.apply(circle.getLat()),
+            lonFunction.apply(circle.getLon()),
+            circle.getRadiusMeters()
+        );
+    }
+
+    static double[] quantizeLats(double[] lats, DoubleFunction function) {
+        return Arrays.stream(lats).map(function::apply).toArray();
```

Comment:
I presume we could do a small optimization here if `quantize==false` we could skip the entire stream...toArray.

For example, make two classes, one with two methods that simply return the passed in arrays, and another with two methods that do quantization. Then we initialize those at the top of the main method.